native/packages/react-native-bpk-component-picker/src/BpkPicker.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { TouchableOpacity, View, ViewPropTypes } from 'react-native';
+import PropTypes from 'prop-types';
+import BpkPickerMenu from './BpkPickerMenu';
+
+class BpkPicker extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      pickerOpen: false,
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.ref) {
+      this.props.ref(this);
+    }
+  }
+
+  showPicker = () => {
+    this.setState({
+      pickerOpen: true,
+    });
+    if (this.props.onFocus) {
+      this.props.onFocus();
+    }
+  };
+
+  hidePicker = () => {
+    this.setState({
+      pickerOpen: false,
+    });
+    if (this.props.onBlur) {
+      this.props.onBlur();
+    }
+  };
+
+  render() {
+    const {
+      children,
+      doneLabel,
+      onBlur,
+      onFocus,
+      onValueChange,
+      ref,
+      renderTrigger,
+      selectedValue,
+      style,
+      ...rest
+    } = this.props;
+
+    const accessibilityTraits = ['button'];
+
+    return (
+      <View style={style}>
+        <TouchableOpacity
+          accessibilityComponentType="button"
+          accessibilityTraits={accessibilityTraits}
+          onPress={this.showPicker}
+        >
+          {renderTrigger && renderTrigger(selectedValue)}
+        </TouchableOpacity>
+        <BpkPickerMenu
+          {...rest}
+          doneLabel={doneLabel}
+          onClose={() => this.hidePicker()}
+          onValueChange={onValueChange}
+          selectedValue={selectedValue}
+          visible={this.state.pickerOpen}
+        >
+          {children}
+        </BpkPickerMenu>
+      </View>
+    );
+  }
+}
+
+BpkPicker.propTypes = {
+  children: PropTypes.node.isRequired,
```

Comment:
Lots of duplication here between components. This will be addressed in a follow-up PR.